tests/testing_utils.py

```diff
@@ -0,0 +1,44 @@
+import glob
+
+import subprocess
+
+
+def run_snakemake(env_variables, options=['-j', '4'], target=''):
+    cmd_args = ["snakemake"] + options
+    if target:
+        cmd_args.append(target)
+    subprocess.call(cmd_args, env=env_variables)
+
+
+def gzipped_result_dir_same_as_reference(result, reference):
+    """
+    Return True if all gzipped files in result directory matched the gzipped files in
+    the reference directory and False otherwise.
+    """
+    result_files = sorted(glob.glob(result + '/**/*.gz', recursive=True))
+    reference_files = sorted(glob.glob(reference + '/**/*.gz', recursive=True))
+    return all(
+        gzipped_result_file_same_as_reference(res, ref)
+        for res, ref in zip(result_files, reference_files)
+    )
+
+
+def gzipped_result_file_same_as_reference(result, reference):
+    """
+    Return True if a gzipped file result matches the gzipped reference file and False
+    otherwise. Do not compare a header of a gzipped file, which contains information
+    on when the file was compressed.
+    """
+    cmd_args = ['cmp', '-i', '8']  # skip a header
+    code = subprocess.call(cmd_args + [result, reference])
+    return not bool(code)
```

Comment:
I forget whether subprocess makes this feasible, but is it possible to raise an error with the contents of stderr in it, in the case where the return code is non-zero, instead of just returning False?

Reply:
So right now we'd still print out which files differ and where. Do I understand correctly that you want more information than that? And if so, what information?